src/Distributed/distributed_grids.jl

```diff
@@ -1,19 +1,30 @@
 using MPI
+using Oceananigans.Utils: getname
 using Oceananigans.Grids: topology, size, halo_size, architecture, pop_flat_elements
 using Oceananigans.Grids: validate_rectilinear_grid_args, validate_lat_lon_grid_args
 using Oceananigans.Grids: generate_coordinate, with_precomputed_metrics
 using Oceananigans.Grids: cpu_face_constructor_x, cpu_face_constructor_y, cpu_face_constructor_z
+using Oceananigans.Grids: R_Earth, metrics_precomputed
+
+using Oceananigans.ImmersedBoundaries
 
 import Oceananigans.Grids: RectilinearGrid, LatitudeLongitudeGrid, with_halo
 
 const DistributedGrid{FT, TX, TY, TZ} = AbstractGrid{FT, TX, TY, TZ, <:MultiArch}
+const DistributedRectilinearGrid{FT, TX, TY, TZ, FX, FY, FZ, VX, VY, VZ} =
+    RectilinearGrid{FT, TX, TY, TZ, FX, FY, FZ, VX, VY, VZ, <:MultiArch} where {FT, TX, TY, TZ, FX, FY, FZ, VX, VY, VZ}
```

Comment:
Should the type of the architecture be moved forward among the type parameters to make this more readable?